src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializerFactory.java

```diff
@@ -670,14 +677,23 @@ protected void addInjectables(DeserializationContext ctxt,
      * has been designated as such setter.
      */
     protected SettableAnyProperty constructAnySetter(DeserializationContext ctxt,
-            BeanDescription beanDesc, AnnotatedMethod setter)
+            BeanDescription beanDesc, AnnotatedMember setter)
         throws JsonMappingException
     {
         if (ctxt.canOverrideAccessModifiers()) {
             setter.fixAccess(ctxt.isEnabled(MapperFeature.OVERRIDE_PUBLIC_ACCESS_MODIFIERS)); // to ensure we can call it
         }
-        // we know it's a 2-arg method, second arg is the value
-        JavaType type = setter.getParameterType(1);
+
+        //find the java type based on the annotated setter method or setter field 
+        JavaType type = null;
+		if (setter instanceof AnnotatedMethod) {
+			// we know it's a 2-arg method, second arg is the value
+			type = ((AnnotatedMethod) setter).getParameterType(1);
+		} else if (setter instanceof AnnotatedField) {
+			// get the type from the content type of the map object
```

Comment:
This would be good place to verify that we got a `Map` valued property; type should then be `MapType` as well (and guarantee there is content type to use).